server/service/src/contact_form/insert/validate.rs

```diff
@@ -0,0 +1,37 @@
+use regex::Regex;
+use repository::{contact_form_row::ContactFormRowRepository, RepositoryError, StorageConnection};
+
+use super::{InsertContactForm, InsertContactFormError};
+
+pub fn validate(
+    input: &InsertContactForm,
+    connection: &StorageConnection,
+) -> Result<(), InsertContactFormError> {
+    if check_contact_form_record_exists(&input.id, connection)? {
+        return Err(InsertContactFormError::ContactIdAlreadyExists);
+    }
+
+    if input.reply_email.is_empty() {
+        return Err(InsertContactFormError::EmailDoesNotExist);
+    }
+
+    let email_regex = Regex::new(r"^[\w\.=-]+@[\w\.-]+\.[\w]{2,}$").unwrap();
```

Comment:
This regex will fail on an address like this `james+test@gmail.com` which is a valid email.
If we're going to have a email regex in open-mSupply I think it should have tests associated with in, and probably live in the `utils` package.

Validating email addresses is difficult, and I think we're generally better to be super conservative and then just let the email fail to send if it's invalid? 

In notify currently we're using this in the frontend, I think a frontend check is fine here for this purpose...

https://github.com/msupply-foundation/notify/blob/e97525854006a6e247cf34b9aef5222c37891cf3/frontend/packages/common/src/utils/regex/RegexUtils.tsx#L49

If you do want a backend check, I'd potentialy just do this. `^[^@]+@[^@]+$`
That means any characters except for an `@` sign followed by an `@` sign and then any other non-`@` characters.
That basically just ensures we have only one @ sign, so it would accept something like `james @msupply.foundation` which isn't valid but remember this record will be synced so we'll still have the record even if the email is invalid...

Reply:
@jmbrunskill - confirming, when you say "conservative", you mean conservative by erring on the side of accepting the email, rather than conservative preventing potentially invalid emails? 😁 

I was wondering if it was better to try constrain more... as if record syncs but email never sends due to invalid address, how long until that is discovered (particularly if it was a support request that was sent)? I suppose we could recommend configuring notify alerts against mail send failures?

Reply:
> @jmbrunskill - confirming, when you say "conservative", you mean conservative by erring on the side of accepting the email, rather than conservative preventing potentially invalid emails? 😁

Basically yes, even if the email is syntactically valid. it can be miss entered or have typos and might not send anyway, so that situation needs to be accounted for. If I understand correctly, we'll send these to support or some known email address anyway, so it would only be the email to the user that would go missing.

I'm just aware how many times email address haven't been considered valid that should have been, some websites don't accept `msupply.foundation` as a domain for example.

It's just my 2cents though, feel free to try for something more strict, but I'm worried it might create more problems than it solves :)

Also might be worth considering if you can use the user's email from the `user_account` table, pre-populate for them mabye? Not sure how often this is filled in though...

Reply:
Yes have a future todo to pre-populate with user account email if defined 🎉 

Yeah that's true, it should make it to support regardless, which is the main thing 🙏 Thanks for your input James, makes sense, we'll update to be less restrictive :)

Reply:
Great feedback, thank you James. I have updated the regex to ```[^@]+@[^@]+\.[^@]+``` very similar to your suggestion but requiring the .  to get the basic structure in

Would this be a refactor issue to bring the validation into the utils?

Reply:
> Would this be a refactor issue to bring the validation into the utils?

It wouldn't hurt to create refactor issue for it...